pkg/userdata/flatcar/provider.go

```diff
@@ -33,6 +32,12 @@ import (
 	userdatahelper "github.com/kubermatic/machine-controller/pkg/userdata/helper"
 )
 
+const (
+	poseidonImage    = "quay.io/poseidon/kubelet"
```

Comment:
I would rather call it `kubeletImage` in such case, giving an option to override it could turn useful indeed

Reply:
then we'd need 2 flags, 1 for "< 1.18" kubelets and 1 for "> 1.18".

Reply:
I think we could have just one that applies to both cases. We could default to poseidon for "> 1.18" and hyperkube for "< 1.18", but again I don't think this is fundamental.

Reply:
> I think we could have just one that applies to both cases. We could default to poseidon for "> 1.18" and hyperkube for "< 1.18", but again I don't think this is fundamental.

The problem with this approach is that someone can use MC to create both > 1.18 and < 1.18 nodes at the same time, so I'm not sure that one flag would work. I think we can have just one flag for Poseidon if someone wants to use their own image instead of a third-party one, and always use the upstream image for Hyperkube.

On the other hand, I don't think this is very important, so we can skip it unless there's no an request for this.

Reply:
> The problem with this approach is that someone can use MC to create both > 1.18 and < 1.18 nodes at the same time, so I'm not sure that one flag would work. I think we can have just one flag for Poseidon if someone wants to use their own image instead of a third-party one, and always use the upstream image for Hyperkube.

makes sense, anyway this flag could be useful for scenarios with limited internet access, thus it's probably better to have it from the beginning. As you pointed out we have already a flag for `< 1.18` so we could add something like `kubeletRepository` for `>= 1.18`

Reply:
In addition to the `-node-hyperkube-image`, there is new flag `-node-kubelet-image`, which regulates the second image. Someday we will deprecate and remove `-node-hyperkube-image`, but for now looks like we need them both.

Reply:
Sorry for the cosmetic comment, but wouldn't `kubelet-repository` be a better name? `node-` seems superfluous and `repository` is more precise as the tag shouldn't be specified.

Reply:
I believe that `node-kubelet-repository` would be a better name because it matches other related flags.

Reply:
@kron4eg I think this is the last point, don't you think that `repository` would be more appropriate than `image`? It would avoid you to specify that the tag should not be specified.

Reply:
the `-node-*` family of flags inherit this `node` prefix to signify that those settings directly control what will land at the node.

```
  -node-http-proxy string
        If set, it configures the 'HTTP_PROXY' & 'HTTPS_PROXY' environment variable on the nodes.
  -node-hyperkube-image string
        Image for the hyperkube container excluding tag. Only has effect on CoreOS Container Linux and Flatcar Linux, and for kubernetes < 1.18. (default "k8s.gcr.io/hyperkube-amd64")
  -node-insecure-registries string
        Comma separated list of registries which should be configured as insecure on the container runtime
  -node-kubelet-image string
        Image for the kubelet container excluding tag. Only has effect on Flatcar Linux, and for kubernetes >= 1.18. (default "quay.io/poseidon/kubelet")
  -node-no-proxy string
        If set, it configures the 'NO_PROXY' environment variable on the nodes. (default ".svc,.cluster.local,localhost,127.0.0.1")
  -node-pause-image string
        Image for the pause container including tag. If not set, the kubelet default will be used: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/
  -node-registry-mirrors string
```

so we ether review this "prefix" grouping or leave `-node-kubelet-image` alone.

Reply:
@kron4eg The prefix is okay. I think @irozzo-1A is talking about renaming the flag to `-node-kubelet-repository`, i.e. replace `image` with `repository`.

Reply:
yep, I'm not fond of this `node-` prefix but lets keep it for consistency. The proposal was to have `node-kubelet-repository` instead of `node-kubelet-image` as the term `repository` looks more appropriated.

Reply:
Renamed